pkgs/development/compilers/sac2c/sac2c-unwrapped.nix

```diff
@@ -0,0 +1,64 @@
+{ lib, pkgs, stdenv, fetchgit,
+  glibc, libuuid, libxcrypt,
+  libxslt, gnum4, python3,
+  gcc, binutils, bash,
+  autoPatchelfHook,
+  cudaPackages, hwloc,
+  cmake
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sac2c-unwrapped";
+
+  version-main = "v1.3.3";
+  version-name = "MijasCosta";
+  version-commit = "1026";
+  version-rev = "g3af2";
+
+  version = "${version-main}-${version-name}-${version-commit}-${version-rev}";
+
+  src = fetchgit {
+    url = "https://gitlab.sac-home.org/sac-group/sac2c.git";
+    rev = "3af2fd45b0ba04e3150d89ae20c92e9ec143dc2c";
+    sha256 = "sha256-2GUW+SqbD9jr+ctFPs1WCj2VvK7XIQZS263SFHO+FHQ=";
+  };
+
+  patches = [
+    ./sac2crc.patch
+  ];
+
+  prePatch = ''
+    substituteInPlace cmake/check-repo-version.cmake \
+      --replace "\''${GIT_COMMAND} describe --tags --abbrev=4 --dirty" "echo ${version}"
```

Comment:
Pushing a programmatic solution upstream would be cool. We do assume that any build happens from a git repo, such that we can derive a _version_ from the history, this however is limits build options. Maybe introduce a mechanism for provide a source release?